src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -191,6 +193,34 @@ public static Appointment parseAppointment(String appointmentName, String appoin
         return new Appointment(trimmedAppointmentName, appointmentDate);
     }
 
+    /**
+     * Parses a {@code String date} into a {@code LocalDate}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code date} is invalid.
+     */
+    public static LocalDate parseDate(String date) throws ParseException {
+        requireNonNull(date);
+        String trimmedDate = date.trim();
+
+        String dateValidation = "\\d{2}-\\d{2}-\\d{4}";
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+
+
+        if (!trimmedDate.matches(dateValidation)) {
+            throw new ParseException(CompleteCommand.MESSAGE_INVALID_DATE_FORMAT);
+        }
```

Comment:
What does this data validation checks if there is already a try and catch block?